modules/tncIdSystem.js

```diff
@@ -0,0 +1,69 @@
+import { submodule } from '../src/hook.js';
+import { logInfo } from '../src/utils.js';
+import { loadExternalScript } from '../src/adloader.js';
+
+const MODULE_NAME = 'tncId';
+
+const FALLBACK_TNC_PROVIDERID = 'c8549079-f149-4529-a34b-3fa91ef257d1';
+const FALLBACK_TNC_INSTANCE = '__tncPbjs';
+
+let disableFallbackScript = false;
+
+const waitTNCScript = (tncNS) => {
+  return new Promise((resolve, reject) => {
+    var tnc = window[tncNS];
+    if (!tnc) reject(new Error('No TNC Object'));
+    if (tnc.tncid) resolve(tnc.tncid);
+    tnc.ready(() => {
+      tnc = window[tncNS];
+      if (tnc.tncid) resolve(tnc.tncid);
+      else tnc.on('data-sent', () => resolve(tnc.tncid));
+    });
+  });
+}
+
+const loadRemoteScript = (providerId) => {
+  return new Promise((resolve) => {
+    loadExternalScript('https://js.tncid.app/remote.js?ns=' + FALLBACK_TNC_INSTANCE + '&providerId=' + providerId, MODULE_NAME, resolve);
```

Comment:
Thank you for converting to load external script, please also do either of the below:

1) open source and lock this script to a version
2) change its location to a configuration parameter with no default ( but feel free to mention the usual location in an integration example or documentation )

Thanks!!

Reply:
Thank you!